packages/flutter_tools/lib/src/drive/web_driver_service.dart

```diff
@@ -78,6 +78,7 @@ class WebDriverService extends DriverService {
           hostname: debuggingOptions.hostname,
           webRenderer: debuggingOptions.webRenderer,
           webUseWasm: debuggingOptions.webUseWasm,
+          webHeaders: debuggingOptions.webHeaders,
```

Comment:
This change is going to need a test, that verifies that any headers that are passed to the web runner's debugging options are correct. Unfortunately https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/web.shard/web_driver_service_test.dart only contains a single test, which uses `startTest()` and not the `start()` method of the service

Reply:
What do you think about adding the test to https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/general.shard/drive/web_driver_service_test.dart since it already contains other tests that validate the behavior of the `start()` method of the service?

I have added a test there to validate that correct webHeaders are passed.

Reply:
That test seems reasonable to me!